server/controllers/profileControllers.js

```diff
@@ -0,0 +1,76 @@
+const createError = require("http-errors");
+const Profile = require('../models/profileModel');
+
+/**
+ * CREATE /profile
+ * - Given parameters passed in, create a profile. 
+ */
+exports.create = async (req, res, next) => {
+    try {
+        const { firstName, lastName, gender, birthDate, description } = req.body;
+        //check for first and last name 
+        if (!firstName) {
+            return next(createError(400, "First name required!"));
+        }
+        if (!lastName) {
+            return next(createError(400, "Last name required!"));
+        }
+
+        //create new profile
+        const newProfile = new Profile({
+            firstName,
+            lastName,
+            gender,
+            birthDate,
+            description
+        });
+        const savedProfile = await newProfile.save();
+
+        res.status(200).send(savedProfile);
+    }catch (err) {
+        next(createError(500, err.message));
+    }
+}
+/** 
+ * UPDATE /profile/:id 
+ * - Given an ID and new parameters, update the profile
+*/
+exports.updateProfile = async (req, res, next) => {
+    try{
+        //retrieve profile and update necessary fields
+        const updatedProfile = await Profile.findOneAndUpdate(
```

Comment:
It would be a good idea to add a 404 error if no profile is found (like you have for the GET route)